flow/connectors/clickhouse/clickhouse.go

```diff
@@ -0,0 +1,128 @@
+package connclickhouse
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log/slog"
+	"strings"
+
+	_ "github.com/ClickHouse/clickhouse-go/v2"
+	_ "github.com/ClickHouse/clickhouse-go/v2/lib/driver"
+
+	//_ "github.com/ClickHouse/clickhouse-go"
+
+	"github.com/PeerDB-io/peer-flow/generated/protos"
+	"github.com/PeerDB-io/peer-flow/shared"
+)
+
+//nolint:stylecheck
+type tableNameComponents struct {
+	schemaIdentifier string
+	tableIdentifier  string
+}
+
+type ClickhouseConnector struct {
+	ctx                context.Context
+	database           *sql.DB
+	tableSchemaMapping map[string]*protos.TableSchema
+	logger             slog.Logger
+}
+
+// creating this to capture array results from clicknhouse.
```

Comment:
you mentioned array support not being ready, can this be added as a part of array support MR?